tensorflow/compiler/xla/service/all_reduce_reassociate.cc

```diff
@@ -57,16 +108,31 @@ StatusOr<bool> AllReduceReassociate::Run(
   bool changed = false;
   for (auto computation : module->computations(execution_threads)) {
     for (HloInstruction *inst : computation->MakeInstructionPostOrder()) {
+      HloInstruction *convert0 = nullptr;
+      HloInstruction *convert1 = nullptr;
       std::optional<ReductionKind> kind = MatchReductionInstruction(inst);
-      if (!kind || inst->operand(0)->opcode() != HloOpcode::kAllReduce ||
-          inst->operand(1)->opcode() != HloOpcode::kAllReduce ||
-          !inst->shape().IsArray()) {
+      if (!kind || !MatchAllreduceOperandsWithOptionalConvert(inst, &convert0,
+                                                              &convert1)) {
+        continue;
+      }
+      // Check to see if input converts are casting to different types,
```

Comment:
update comment: it seems this checks if inputs converts are value preserving and if both are present, compatible with each other. Update name and VLOG(2) message to reflect that?

Reply:
updated